lib/reek/cli/options.rb

```diff
@@ -213,6 +214,15 @@ def set_utility_options
           puts parser
           exit
         end
+        parser.on_tail('-l', '--list', 'List all available smell detectors') do
+          puts 'All available smell detectors:\n'
+          DetectorRepository.smell_types.map do |full_name|
```

Comment:
For improved testability, maybe move the generation of the list to another class?

Reply:
How about creating

>> DetectorRepository#print_available_detectors

?

Reply:
Perhaps `DetectorRepository.available_detector_names`, so you can do `puts DetectorRepository.available_detector_names`?

Reply:
Yeah, good idea. I'd suggest to also rename `DetectorRepository.smell_types` to `detectors` since the method returns ..... detectors and not smell_types. We kind of fucked up our naming a bit here and there ;)

Reply:
Yes, that makes sense as well :-).

Reply:
Fixed. I'll fix DetectorRepository.smell_types in another pull request so I don't blow up this one.